dot15d4/src/phy/device.rs

```diff
@@ -0,0 +1,15 @@
+use crate::time::Instant;
+
+/// An IEEE 802.15.4 device.
+pub trait Device {
+    fn disable(&mut self);
+    fn enable(&mut self);
+
+    fn receive<RX>(&mut self, rx: RX)
+    where
+        RX: FnMut(&[u8], Instant);
+
+    fn transmit<TX>(&mut self, tx: TX)
+    where
+        TX: for<'b> Fn(&'b mut [u8]) -> Option<&'b [u8]>;
+}
```

Comment:
Could/Shouldn't you add documentation to these functions as well?

Reply:
This is actually not used by the CSMA implementation. I kept it because it was already defined, and moved it to more (at the moment of writing) a more sensible place. Should this be removed?

The idea is more to use the `Driver` trait as an interface with higher layers. This is because (eg. in Embassy), the logic that owns the radio (our case) often runs in a separate task that communicates over channels.